tests/test_govuk_frontend.py

```diff
@@ -65,6 +65,122 @@ def test_with_errors(self, question, snapshot):
         assert from_question(question, errors=errors) == snapshot
 
 
+class TestNumberInput:
+    @pytest.fixture
+    def question(self):
+        return Question({
+            "id": "howMany",
+            "question": "How many?",
+            "type": "number",
+        })
+
+    def test_govuk_input(self, question, snapshot):
+        assert govuk_input(question) == snapshot
+
+    def test_govuk_input_asking_for_whole_numbers(self, question):
+        """Test that we follow Design System guidance on asking for numbers
+
+        https://design-system.service.gov.uk/components/text-input/#numbers
+        """
+
+        question.limits = {"integer_only": True}
+
+        params = govuk_input(question)
+
+        assert params["inputmode"] == "numeric"
+        assert params["pattern"] == "[0-9]*"
+        assert params["spellcheck"] is False
+
+    @pytest.mark.parametrize("integer_only", (False, None))
+    def test_govuk_input_asking_for_decimal_numbers(self, integer_only, question):
+        """Test that we follow Design System guidance on asking for numbers
+
+        https://design-system.service.gov.uk/components/text-input/#numbers
+        """
+
+        if integer_only is False:
+            question.limits = {"integer_only": False}
+
+        params = govuk_input(question)
+
+        assert params["inputmode"] == "text"
```

Comment:
Decimal numbers should not include `inputmode`

https://design-system.service.gov.uk/components/text-input/#asking-for-decimal-numbers

Reply:
Ah right I misread that! Good spot, I'll fix that.

Reply:
Fixed